edit_questionpy_form.php

```diff
@@ -203,7 +204,7 @@ private function definition_package_settings_search(MoodleQuickForm $mform) {
             $isfavourite = $ufservice->favourite_exists('qtype_questionpy', 'package', $package->id, $usercontext);
             $version = $pkgversion->version;
         } else {
-            $api = new api();
+            $api = new api(new qpy_http_client());
             $package = $api->get_package_info($packagehash);
```

Comment:
Bei mir ist die Klasse vor ein paar Tagen auch aufgeschlagen, weil Property-Deklarationen für `$api` und `$packagefileservice` fehlen, obwohl sie im Construktor gesetzt werden. Magst du die direkt nachziehen und dann:

```suggestion
            $this->api->get_package_info($packagehash);
```
?

Reply:
Siehe <https://github.com/questionpy-org/moodle-qtype_questionpy/commit/3c4b1241ee2d37cc142dd0fd1cdbcb4e9b794910#diff-c0497fb20dfbffd0664ba3b05fee8464f83df7b0e5ed4e6bda097ae9d1fc8a16R44-R48>